great_expectations/datasource/fluent/data_asset/data_connector/file_path_data_connector.py

```diff
@@ -149,15 +149,15 @@ def get_data_reference_count(self) -> int:
 
     # Interface Method
     @override
-    def get_matched_data_references(self) -> List[str]:
+    def get_matched_data_references(self, regex: re.Pattern | None = None) -> List[str]:
```

Comment:
this change is required to test if a path exists using our current structure. In the world of BatchDefinitions it probably makes sense to update the API across the board to require a batch request here, but i wanted to minimize scope for this PR.

Reply:
I see below that you are doing an existence test with a passed in regex. I don't know the flow but that use case makes sense.

Reply:
the new pattern is that the `DataConnector` gets the regex from the batch request. We haven't removed `batching_regex` from the asset init signature yet (which passes it through the DataConnector init signature as well), so we prefer the regex via args if possible, but if it isn't provided then we're in a legacy workflow and we default to the regex saved on the class. The class level regex should be removed soon!